packages/react-scheduler/src/ReactScheduler.js

```diff
@@ -154,11 +151,14 @@ if (!ExecutionEnvironment.canUseDOM) {
       didTimeout = false;
     }
 
-    const callback = scheduledRICCallback;
-    resetScheduledCallback();
+    const callback = scheduledCallback;
+    timeoutTime = -1;
+    scheduledCallback = null;
     if (callback !== null) {
       frameDeadlineObject.didTimeout = didTimeout;
+      isCurrentlyRunningCallback = true;
       callback(frameDeadlineObject);
+      isCurrentlyRunningCallback = false;
```

Comment:
This callback could throw, in which case `isCurrentlyRunningCallback` won't be reset.